### DISCUSSION TITLE
How do I use Github Copilot with `lsp-zero`?

### DISCUSSION THREAD

**Discussion by githorse** [VonHeikemen/lsp-zero.nvim#352] [Q&A]
I've set up `lsp-zero` according to the recommendations, including this in `packer.lua`:

```
use {
	'VonHeikemen/lsp-zero.nvim',
	branch = 'v1.x',
	requires = {

		-- LSP Support
		{'neovim/nvim-lspconfig'},             -- Required
		{'williamboman/mason.nvim'},           -- Optional
		{'williamboman/mason-lspconfig.nvim'}, -- Optional

		-- Autocompletion
		{'hrsh7th/nvim-cmp'},         -- Required
		{'hrsh7th/cmp-nvim-lsp'},     -- Required
		{'hrsh7th/cmp-buffer'},       -- Optional
		{'hrsh7th/cmp-path'},         -- Optional
		{'saadparwaiz1/cmp_luasnip'}, -- Optional
		{'hrsh7th/cmp-nvim-lua'},     -- Optional

		-- Snippets
		{'L3MON4D3/LuaSnip'},             -- Required
		{'rafamadriz/friendly-snippets'}, -- Optional
	}
}
```

Now I've followed the instructions to install and enable [Github Copilot for Neovim](https://github.com/github/copilot.vim). Out of the box, `:Copilot status` tells me that `<Tab> map has been disabled or is claimed by another plugin`. Attempts to [debug that](https://www.reddit.com/r/neovim/comments/sk70rk/using_github_copilot_in_neovim_tab_map_has_been/) are so far unsuccesful.

Ideally I'd like <Tab> completion to work seamlessly with Copilot and other autocompletion sources. But maybe I don't even need those when I'm running Copilot? Looking for advice here. Has anyone successfully gotten Copilot working with `lsp-zero`?

**Comment by VonHeikemen** [comment]
First thing I suggest is updating lsp-zero's version: Change the branch from `v1.x` to `v3.x`.

Next step is to delete whatever configuration you have for lsp-zero right now, and replace it with this:

```lua
local lsp_zero = require('lsp-zero')

lsp_zero.on_attach(function(client, bufnr)
  -- see :help lsp-zero-keybindings
  -- to learn the available actions
  lsp_zero.default_keymaps({buffer = bufnr})
end)

-- see :help lsp-zero-guide:integrate-with-mason-nvim
-- to learn how to use mason.nvim with lsp-zero
require('mason').setup({})
require('mason-lspconfig').setup({
  handlers = {
    lsp_zero.default_setup,
  }
})
```

This particular config does not offer the same set of features as `v1.x` had enabled by default. But copilot should work with it.

if you want to re-enable some of the features from `v1.x`, read this guide: [Migrating from v1 branch](https://github.com/VonHeikemen/lsp-zero.nvim/blob/v3.x/doc/md/guides/migrate-from-v1-branch.md)
